ion/integration_test.go

```diff
@@ -529,13 +529,43 @@ func writeToWriterFromReader(t *testing.T, reader Reader, writer Writer) {
 			}
 
 		case IntType:
-			val, err := reader.BigIntValue()
+			intSize, err := reader.IntSize()
 			if err != nil {
-				t.Errorf("Something went wrong when reading Int value. " + err.Error())
+				t.Errorf("Something went wrong when retrieving the Int size. " + err.Error())
 			}
-			err = writer.WriteBigInt(val)
-			if err != nil {
-				t.Errorf("Something went wrong when writing Int value. " + err.Error())
+
+			switch intSize {
+			case Int32, Int64:
+				val, err := reader.Int64Value()
+				if err != nil {
+					t.Errorf("Something went wrong when reading Int value. " + err.Error())
```

Comment:
Suggested tweak for this and other error messages:

```suggestion
					t.Errorf("Something went wrong while reading an Int value: " + err.Error())
```